rpc/monitor.go

```diff
@@ -0,0 +1,36 @@
+package rpc
+
+import (
+	"net/http"
+)
+
+func (h *Handler) HandleReadyRequest(w http.ResponseWriter, r *http.Request) {
+	if h.ready {
+		w.WriteHeader(http.StatusOK)
+	} else {
+		w.WriteHeader(http.StatusServiceUnavailable)
+	}
+}
+
+func (h *Handler) HandleReadySyncRequest(w http.ResponseWriter, r *http.Request) {
+	if !h.ready {
+		w.WriteHeader(http.StatusServiceUnavailable)
+	}
+
+	head, err := h.bcReader.HeadsHeader()
+	if err != nil {
+		w.WriteHeader(http.StatusServiceUnavailable)
+		return
+	}
+	highestBlockHeader := h.syncReader.HighestBlockHeader()
+	if highestBlockHeader == nil {
+		w.WriteHeader(http.StatusServiceUnavailable)
+		return
+	}
+
+	if head.Number+6 >= highestBlockHeader.Number {
```

Comment:
let's move magic number (`6`) to a package level constant in this file

Reply:
moved in the new commit